### PR TITLE
FACES-2499 Refactor bridge body renderer to wrap other body renderers

### PR DESCRIPTION
**Pull request by stiemannkj1**

### CODE REVIEW

File: bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+			// Add a special CSS class name, BodyRendererBridgeImpl.STYLE_CLASS_PORTLET_BODY, in the rendered markup
+			// in order to clue-in the developer that a <div> was rendered instead of <body> if the styleClass does not
+			// already contain BodyRendererBridgeImpl.STYLE_CLASS_PORTLET_BODY.
+			if (!valueString.contains(BodyRendererBridgeImpl.STYLE_CLASS_PORTLET_BODY)) {
+				valueString = valueString + " " + BodyRendererBridgeImpl.STYLE_CLASS_PORTLET_BODY;
+			}
+
+			super.writeAttribute(name, valueString, property);
+		}
```
Comment (stiemannkj1) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java]: @ngriffin7a, the below code may be unnecessary, but I noticed that previously, you would loop through [`BODY_PASS_THRU_ATTRIBUTES`](https://github.com/liferay/liferay-faces-bridge-impl/blob/master/bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/BodyRendererBridgeImpl.java#L54) and render them on the `<div>`. Now the wrapped body renderer will do that for us, but [some of the attributes are illegal on `<div>`s (they are meant for a `<body>` tag)](http://www.w3.org/TR/html401/struct/global.html#h-7.5.4). So I suppress them.

File: bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		else if ("onload".equals(name) || "onunload".equals(name) || "role".equals(name) || "xmlns".equals(name)) {
+			logger.warn("Suppressed writing of illegal attribute {0} on portlet body <div>.", name);
+		}
```
Comment (stiemannkj1) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java:86]: @ngriffin7a, the above code may be unnecessary, but I noticed that previously, you would loop through [`BODY_PASS_THRU_ATTRIBUTES`](https://github.com/liferay/liferay-faces-bridge-impl/blob/master/bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/BodyRendererBridgeImpl.java#L54) and render them on the `<div>`. Now the wrapped body renderer will do that for us, but [some of the attributes are illegal on `<div>`s (they are meant for a `<body>` tag)](http://www.w3.org/TR/html401/struct/global.html#h-7.5.4). So I suppress them.

**Comment by ngriffin7a** [comment]
Merged into master (5.0.x) and backported to 4.0.x and 3.0.x (but not 2.0.x since BodyRenderer does not exist in JSF 1.2)

File: bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java (hunk 2 above)
Comment (ngriffin7a) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java:86]: @stiemannkj1 Please set a breakpoint in the codeblock and see if the code is necessary. My best guess is that it still is.

Comment (stiemannkj1) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java:86]: @ngriffin7a,
The code definitely gets hit, but I was asking: "Is this code necessary?" Is it overkill to suppress illegal attributes on the portlet body `<div>`?

Comment (ngriffin7a) [inline bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterBridgeBodyImpl.java:86]: @stiemannkj1 I think it's a good idea to suppress attributes that are illegal on the `<div>`. Thanks.

**Pull request closed without merging**